src/renderer/hooks/store.js

```diff
@@ -12,15 +12,20 @@ const storage = {
    * @returns {string | null}
    */
   getItem: key => {
-    return /** @type {string | null} */ (store.get(key)) || null
+    const item = store.get(key)
+    return item ? JSON.stringify(item) : null
   },
   /**
    * @param {string} key - A string key to reference the stored item by.
    * @param {string} state - The state to be persisted - stringified JSON.
    * @returns {void}
    */
   setItem: (key, state) => {
-    store.set(key, JSON.parse(state))
+    if (typeof state === 'string') {
+      store.set(key, JSON.parse(state))
+    } else {
+      store.set(key, state)
+    }
```

Comment:
with some more testing, it seems like zustand will automatically create your object into a string (so you don't need to do this typecheck)